src/test/shell/bazel/bazel_symlink_test.sh

```diff
@@ -250,7 +314,42 @@ EOF
   [[ "$?" == 1 ]] || fail "unexpected exit code"
 }
 
+function test_symlink_instead_of_file() {
+  mkdir -p a
+  cat > a/a.bzl <<'EOF'
+def _bad_symlink_impl(ctx):
+    target = ctx.actions.declare_file(ctx.label.name + ".target")
+    ctx.actions.write(
+        output = target,
+        content = "Hello, World!",
+    )
+
+    symlink = ctx.actions.declare_symlink(ctx.label.name + ".link")
+    ctx.actions.symlink(
+        output = symlink,
+        target_file = target,
+    )
+    return DefaultInfo(files = depset([symlink]))
+
+bad_symlink = rule(implementation = _bad_symlink_impl)
+EOF
+
+  cat > a/BUILD <<'EOF'
+load(":a.bzl", "bad_symlink")
+
+bad_symlink(name="bs")
+EOF
+
+  bazel build //a:bs && fail "build succeeded"
+  [[ "$?" == 1 ]] || fail "unexpected exit code"
```

Comment:
See above -- instead of cargo-culting the exit code check, use the `$TEST_log` (same for all other tests added by this PR). See other test suites in this directory for examples of how to do it.